core/identity-management/iin-agent/src/common/utils.ts

```diff
@@ -14,3 +30,238 @@ export function handlePromise<T>(promise: Promise<T>): Promise<[T?, Error?]> {
         .catch((error) => Promise.resolve([undefined, error]));
     return result;
 }
+
+export function getIINAgentClient(securityDomain: string, participantId: string, securityDomainDNS?: Object): agent_grpc_pb.IINAgentClient {
+    if (!securityDomainDNS) {
+        securityDomainDNS = getSecurityDomainDNS(securityDomain);
+    }
+    if (!(participantId in securityDomainDNS)) {
+        throw new Error(`DNS for member: ${participantId} of ${securityDomain} not defined in DNS Config`)
+    }
+    const iinAgent = securityDomainDNS[participantId];
+    let client: agent_grpc_pb.IINAgentClient;
+    if (iinAgent.tls === 'true') {
+        if (!iinAgent.tlsCACertPath || iinAgent.tlsCACertPath == "") {
+            client = new agent_grpc_pb.IINAgentClient(
+                iinAgent.endpoint,
+                credentials.createSsl()
+            );
+        } else {
+            if (!(iinAgent.tlsCACertPath && fs.existsSync(iinAgent.tlsCACertPath))) {
+                throw new Error("Missing or invalid IIN Agent's tlsCACertPaths: " + iinAgent.tlsCACertPath);
+            }
+            const rootCert = fs.readFileSync(iinAgent.tlsCACertPath);
+            client = new agent_grpc_pb.IINAgentClient(
+                iinAgent.endpoint,
+                credentials.createSsl(rootCert)
+            );
+        }
+    } else {
+        client = new agent_grpc_pb.IINAgentClient(
+            iinAgent.endpoint,
+            credentials.createInsecure()
+        );
+    }
+    return client;
+}
+
+export function getLedgerBase(securityDomain: string, memberId: string): LedgerBase {
+    const ledgerId = getLedgerId(securityDomain);
+    if(!process.env.DLT_TYPE) {
+        throw new Error(`Env DLT_TYPE not defined`);
+    }
+    const dltType = process.env.DLT_TYPE!.toLowerCase();
+    if(dltType == 'fabric') {
+        const ledgerBase = new FabricConnector(ledgerId, process.env.WEAVER_CONTRACT_ID, securityDomain, process.env.CONFIG_PATH);
+        if (ledgerBase.orgMspId != memberId) {
```

Comment:
Why do we pass `memberId` as parameter if its only purpose is to pass this check? Or, this parameter ought to be passed to the constructor too, so that the returned value stored in `ledgerBase` is a connection to the Fabric ledger for that particular org.

Reply:
We can not enforce connecting to particular org by `memberId`, it depends upon the connection profile.
The memberId is passed yes just to make sure that this is correct (intended) iin-agent.

E.g.: This is used when iin-agent attests membership requested by foreign network, so just to be sure it is actually the attester that requestor requested.

Reply:
There's a redundancy here. We specify the MSP ID in both the `config.json` and in the `.env` and then try to cross-check them here, which serves no purpose. Can we specify it just in one place?

Reply:
Okay can do that, but I didn't made this way to cross-check `.env` with `config.json` instead,
I did this because in case when this iin-agent receives `requestIdentity` request from remote network, it just passes the received `memberId` to this function, and this function checks whether received `memberId` in parameter is same as in config.json. 

Right now checking `memberId` is ledger depenedent i.e. orgMspId for `FabricConnector` something else for CordaConnector probably. So that's why to verify that I had it here instead of the place where we call this function (didn't want to put ledger specific switch case there).

I think better way would be to have `memberId` as property of `ledgerBase`, then this above check can easily be done in the respective endpoint function instead of here in utils, what do you think?

Reply:
Yeah, I agree. Member ID is an attribute in our protos, so it's meant to be DLT-neutral. Keeping it in `ledgerBase` should work.

Reply:
Done.